<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Publications</title>
  <meta name="description" content="From data to insight – Putting the human back in the loop of data analysis
">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="canonical" href="http://vda-lab.be/publications.html">
  <link rel="alternate" type="application/rss+xml" title="Visual Data Analysis Lab" href="http://vda-lab.be/feed.xml" />
</head>


  <body>
    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">Visual Data Analysis Lab</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
          <a class="page-link" href="/contact.html">Contact</a>
          
        
          
        
          
          <a class="page-link" href="/index.html">Visual Data Analysis Lab</a>
          
        
          
          <a class="page-link" href="/jobs.html">Jobs</a>
          
        
          
          <a class="page-link" href="/lists.html">Lists</a>
          
        
          
        
          
          <a class="page-link" href="/people.html">People</a>
          
        
          
          <a class="page-link" href="/portfolio.html">Portfolio</a>
          
        
          
          <a class="page-link" href="/posts.html">Posts</a>
          
        
          
          <a class="page-link" href="/publications.html">Publications</a>
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <div class="post">

  <header class="post-header">
    <h1 class="post-title">Publications</h1>
  </header>

  <article class="post-content">
    <h1 id="peer-reviewed-journals-">:: Peer-reviewed journals ::</h1>

<ul>
  <li>Sakai R &amp; Aerts J. Card sorting techniques for domain characterization in problem-driven visualization research. Eurographics Conference on Visualization (EuroVis) 2015.</li>
  <li>Sakai, Ryo, and Jan Aerts. 2014. “Sequence Diversity Diagram for Comparative Analysis of Multiple Sequence Alignments.” BMC Proceedings 8 (Suppl 2 Proceedings of the 3rd Annual Symposium on Biological Data Visualization) (January): S9.</li>
  <li>Pougach, Ksenia, Arnout Voet, Fyodor Kondrashov, Karin Voordeckers, Joaquin F Christiaens, Bianka Baying, Vladimir Benes, et al. 2014. “Duplication of a Promiscuous Transcription Factor Drives the Emergence of a New Regulatory Network.” Nature Communications 5 (January): 4868. doi:10.1038/ncomms5868.</li>
  <li>Sakai, Ryo, and Jan Aerts. 2014. “Sequence Diversity Diagram for Comparative Analysis of Multiple Sequence Alignments.” BMC Proceedings 8 (Suppl 2 Proceedings of the 3rd Annual Symposium on Biologica) (January): S9. doi:10.1186/1753-6561-8-S2-S9.</li>
  <li>Sakai, Ryo, Raf Winand, Toni Verbeiren, Andrew Vande Moere, and Jan Aerts. 2014. “Dendsort: Modular Leaf Ordering Methods for Dendrogram Representations in R.” F1000Research 3 (January): 177. doi:10.12688/f1000research.4784.1.</li>
  <li>Winand, Raf, Kristien Hens, Wybo Dondorp, Guido De Wert, Yves Moreau, Joris Robert Vermeesch, Inge Liebaers, and Jan Aerts. 2014. “In Vitro Screening of Embryos by Whole- Genome Sequencing : Now , in the Future or Never ?” 0 (0): 1–10. doi:10.1093/humrep/deu005.</li>
  <li>Pavlopoulos, G. a., P. Kumar, a. Sifrim, R. Sakai, M. L. Lin, T. Voet, Y. Moreau, and J. Aerts. 2013. “Meander: Visually Exploring the Structural Variome Using Space-Filling Curves.” Nucleic Acids Research (11) (April 19): 1–9. doi:10.1093/nar/gkt254.</li>
  <li>Sakai, Ryo, Matthieu Moisse, Joke Reumers, and Jan Aerts. 2013. “Pipit: Visualizing Functional Impacts of Structural Variations.” Bioinformatics (Oxford, England) 29 (17) (September 1): 2206–7. doi:10.1093/bioinformatics/btt367.</li>
  <li>Sakai, Ryo, Alejandro Sifrim, Andrew Vande Moere, and Jan Aerts. 2013. “TrioVis: A Visualisation Approach for Filtering Genomic Variants of Parent-Child Trios.” Bioinformatics (Oxford, England) (May 8): 2–3. doi:10.1093/bioinformatics/btt267.</li>
  <li>Sifrim, Alejandro, Dusan Popovic, Leon-Charles Tranchevent, Amin Ardeshirdavani, Ryo Sakai, Peter Konings, Joris R Vermeesch, Jan Aerts, Bart De Moor, and Yves Moreau. 2013. “eXtasy: Variant Prioritization by Genomic Data Fusion.” Nature Methods 10 (11) (November): 1083–4. doi:10.1038/nmeth.2656.</li>
  <li>Bartlett, Christopher W, Soo Yeon Cheong, Liping Hou, Jesse Paquette, Pek Yee Lum, Günter Jäger, Florian Battke, et al. 2012. “An eQTL Biological Data Visualization Challenge and Approaches from the Visualization Community.” BMC Bioinformatics 13 Suppl 8 (Suppl 8) (January): S8. doi:10.1186/1471-2105-13-S8-S8.</li>
  <li>Bonnal, Raoul J P, Jan Aerts, George Githinji, Naohisa Goto, Dan Maclean, Chase A Miller, Hiroyuki Mishima, et al. 2012. “Biogem : An Effective Tool Based Approach for Scaling up Open Source Software Development in Bioinformatics.” Genetics: 5–7.</li>
  <li>Johansson, Stefan, Henrik Irgens, Kishan K Chudasama, Janne Molnes, Jan Aerts, Francisco S Roque, Inge Jonassen, et al. 2012. “Exome Sequencing and Genetic Testing for MODY.” PloS One 7 (5) (January): e38050. doi:10.1371/journal.pone.0038050.</li>
  <li>Sifrim, Alejandro, Jeroen Kj Van Houdt, Leon-Charles Tranchevent, Beata Nowakowska, Ryo Sakai, Georgios a Pavlopoulos, Koen Devriendt, Joris R Vermeesch, Yves Moreau, and Jan Aerts. 2012. “Annotate-It: A Swiss-Knife Approach to Annotation, Analysis and Interpretation of Single Nucleotide Variation in Human Disease.” Genome Medicine 4 (9) (September 26): 73. doi:10.1186/gm374.</li>
  <li>Katayama, Toshiaki, Mark D Wilkinson, Rutger Vos, Takeshi Kawashima, Shuichi Kawashima, Mitsuteru Nakao, Yasunori Yamamoto, et al. 2011. “The 2nd DBCLS BioHackathon: Interoperable Bioinformatics Web Services for Integrated Applications.” Journal of Biomedical Semantics 2 (1) (August 2): 4. doi:10.1186/2041-1480-2-4.</li>
  <li>Moschopoulos, Charalampos N, Georgios a Pavlopoulos, Ernesto Iacucci, Jan Aerts, Spiridon Likothanassis, Reinhard Schneider, and Sophia Kossida. 2011. “Which Clustering Algorithm Is Better for Predicting Protein Complexes?” BMC Research Notes 4 (1) (December 20): 549. doi:10.1186/1756-0500-4-549.</li>
  <li>Pavlopoulos, Georgios a, Sean D Hooper, Alejandro Sifrim, Reinhard Schneider, and Jan Aerts. 2011. “Medusa: A Tool for Exploring and Clustering Biological Networks.” BMC Research Notes 4 (1) (January): 384. doi:10.1186/1756-0500-4-384.</li>
  <li>Pavlopoulos, Georgios a, Maria Secrier, Charalampos N Moschopoulos, Theodoros G Soldatos, Sophia Kossida, Jan Aerts, Reinhard Schneider, and Pantelis G Bagos. 2011. “Using Graph Theory to Analyze Biological Networks.” BioData Mining 4 (1): 10. doi:10.1186/1756-0381-4-10.</li>
  <li>Strozzi, Francesco, and Jan Aerts. 2011. “A Ruby API to Query the Ensembl Database for Genomic Features.” Bioinformatics (Oxford, England) (January): 1–2. doi:10.1093/bioinformatics/btr050.</li>
  <li>Conrad, Donald F, Dalila Pinto, Richard Redon, Lars Feuk, Omer Gokcumen, Yujun Zhang, Jan Aerts, et al. 2010. “Origins and Functional Impact of Copy Number Variation in the Human Genome.” Nature 464 (April): 704–712. doi:10.1038/nature08516.</li>
  <li>Goto, N., P. Prins, M. Nakao, R. Bonnal, J. Aerts, and T. Katayama. 2010. “BioRuby: Bioinformatics Software for the Ruby Programming Language.” Bioinformatics 26 (20) (August): 2617–2619. doi:10.1093/bioinformatics/btq475.</li>
  <li>Kalay, Ersan, Gökhan Yigit, Yakup Aslan, Karen E Brown, Esther Pohl, Louise S Bicknell, Hülya Kayserili, et al. 2010. “CEP152 Is a Genome Maintenance Protein Disrupted in Seckel Syndrome.” Nature Genetics 43 (1) (December): 23–26. doi:10.1038/ng.725.</li>
  <li>The DBCLS BioHackathon Consortium. 2010. “The DBCLS BioHackathon: Standardization and Interoperability for Bioinformatics Web Services and Workflows.” Journal of Biomedical Semantics. doi:10.1186/2041-1480-1-8.</li>
  <li>The Wellcome Trust Case Control Consortium. 2010. “Genome-Wide Association Study of CNVs in 16 , 000 Cases of Eight Common Diseases and 3 , 000 Shared Controls.” Nature 464 (7289): 713–720. doi:10.1038/nature08979.</li>
  <li>Aerts, Jan A, and Chris Tyler-Smith. 2009. “Structural Variation in Great Ape Genomes.” In In: Encyclopedia of Life Sciences. doi:10.1002/9780470015902.a0021768.</li>
  <li>Aerts, Jan, and Andy Law. 2009. “An Introduction to Scripting in Ruby for Biologists.” BMC Bioinformatics 10 (January): 221. doi:10.1186/1471-2105-10-221.</li>
  <li>Connelley, Timothy, Jan Aerts, Andy Law, and W Ivan Morrison. 2009. “Genomic Analysis Reveals Extensive Gene Duplication within the Bovine TRB Locus.” BMC Genomics 10 (January): 192. doi:10.1186/1471-2164-10-192.</li>
  <li>Elsik, Christine G, Ross L Tellam, Richard A Gibbs, Kim C Worley, Donna M Muzny, George M Weinstock, David L Adelson, et al. 2009. “The Genome Sequence of Taurine Cattle: A Window to Ruminant Biology and Evolution.” Science 324 (5926): 522–528. doi:10.1126/science.1169588.</li>
  <li>Gibbs, Richard a, Jeremy F Taylor, Curtis P Van Tassell, William Barendse, Kellye A Eversole, Clare A Gill, Ronnie D Green, et al. 2009. “Genome-Wide Survey of SNP Variation Uncovers the Genetic Structure of Cattle Breeds.” Science 324 (5926): 528–532. doi:10.1126/science.1167936.</li>
  <li>McKay, Stephanie D, Robert D Schnabel, Brenda M Murdoch, Lakshmi K Matukumalli, Jan Aerts, Wouter Coppieters, Denny Crews, et al. 2008. “An Assessment of Population Structure in Eight Breeds of Cattle Using a Whole Genome SNP Panel.” BMC Genetics 9 (January): 37. doi:10.1186/1471-2156-9-37.</li>
  <li>Taylor, Chris F, Dawn Field, Susanna-Assunta Sansone, Jan Aerts, Rolf Apweiler, Michael Ashburner, Catherine A Ball, et al. 2008. “Promoting Coherent Minimum Reporting Guidelines for Biological and Biomedical Investigations: The MIBBI Project.” Nature Biotechnology 26 (8) (August): 889–96. doi:10.1038/nbt.1411.</li>
  <li>Aerts, J a, H J Megens, T Veenendaal, I Ovcharenko, R Crooijmans, L Gordon, L Stubbs, and M Groenen. 2007. “Extent of Linkage Disequilibrium in Chicken.” Cytogenetic and Genome Research. doi:10.1159/000103196.</li>
  <li>McKay, S D, R D Schnabel, B M Murdoch, J Aerts, C a Gill, C Gao, C Li, et al. 2007. “Construction of Bovine Whole-Genome Radiation Hybrid and Linkage Maps Using High-Throughput Genotyping.” Animal Genetics 38 (2) (April): 120–5. doi:10.1111/j.1365-2052.2006.01564.x.</li>
  <li>McKay, Stephanie D, Robert D Schnabel, Brenda M Murdoch, Lakshmi K Matukumalli, Jan Aerts, Wouter Coppieters, Denny Crews, et al. 2007. “Whole Genome Linkage Disequilibrium Maps in Cattle.” BMC Genetics 8: 74. doi:10.1186/1471-2156-8-74.</li>
  <li>Snelling, Warren M, Readman Chiu, Jacqueline E Schein, Matthew Hobbs, Colette a Abbey, David L Adelson, Jan Aerts, et al. 2007. “A Physical Map of the Bovine Genome.” Genome Biology 8 (8) (January): R165. doi:10.1186/gb-2007-8-8-r165.</li>
  <li>Jann, Oliver C, Jan Aerts, Michelle Jones, Nicola Hastings, Andy Law, Stephanie McKay, Elisa Marques, et al. 2006. “A Second Generation Radiation Hybrid Map to Aid the Assembly of the Bovine Genome Sequence.” BMC Genomics 7 (January): 283. doi:10.1186/1471-2164-7-283.</li>
  <li>Aerts, J, T Veenendaal, J J van der Poel, R P M a Crooijmans, and M a M Groenen. 2005. “Chromosomal Assignment of Chicken Clone Contigs by Extending the Consensus Linkage Map.” Animal Genetics 36 (3) (June): 216–22. doi:10.1111/j.1365-2052.2005.01289.x.</li>
  <li>Aerts, Jan a, Bart J Jungerius, and Martien a M Groenen. 2004. “POSA: Perl Objects for DNA Sequencing Data Analysis.” BMC Genomics 5 (1) (August): 60. doi:10.1186/1471-2164-5-60.</li>
  <li>International Chicken Genome Sequencing Consortium. 2004. “Sequence and Comparative Analysis of the Chicken Genome Provide Unique Perspectives on Vertebrate Evolution.” Nature 432 (December): 695–777.</li>
  <li>Wallis, John W, Jan Aerts, Martien AM Groenen, Richard PMA Crooijmans, Dan Layman, Tina A Graves, Debra E Scheer, et al. 2004. “A Physical Map of the Chicken Genome.” Nature 432 (December): 761–764. doi:10.1038/nature03030.</li>
  <li>Aerts, J, R Crooijmans, S Cornelissen, K Hemmatian, T Veenendaal, A Jaadar, J Van Der Poel, V Fillon, A Vignal, and M Groenen. 2003. “Integration of Chicken Genomic Resources to Enable Whole-Genome Sequencing.” Cytogenetic and Genome Research 303: 297–303. doi:10.1159/000075766.</li>
  <li>Aerts, Jan, Yves Wetzels, Nadine Cohen, and Jeroen Aerssens. 2002. “Data Mining of Public SNP Databases for the Selection of Intragenic SNPs.” Human Mutation 20 (3) (September): 162–73. doi:10.1002/humu.10107.</li>
</ul>

<h1 id="general-media-">:: General media ::</h1>

<ul>
  <li>De Standaard, 04/06/2014, “Je smartphone als dokter”</li>
  <li>Aerts J. Enhancing genomic analysis through ICT and visualisation. Projects, Insight Publishers, p28-31 (2013)</li>
  <li>Scheire, L. &amp; Van Tendeloo, H. (2012) Wat cellen voorspellen: de genetica van de toekomst. Humo, 3725/04, 10-15</li>
</ul>

  </article>

</div>

      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">

    <h2 class="footer-heading">Visual Data Analysis Lab</h2>

    <div class="footer-col-wrapper">
      <div class="footer-col  footer-col-1">
        <ul class="contact-list">
          <li>Visual Data Analysis Lab</li>
          <li><a href="mailto:jan.aerts@kuleuven.be">jan.aerts@kuleuven.be</a></li>
        </ul>
      </div>

      <div class="footer-col  footer-col-2">
        <ul class="social-media-list">
          

          
          <li>
            <a href="https://twitter.com/jandot">
              <span class="icon  icon--twitter">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M15.969,3.058c-0.586,0.26-1.217,0.436-1.878,0.515c0.675-0.405,1.194-1.045,1.438-1.809
                  c-0.632,0.375-1.332,0.647-2.076,0.793c-0.596-0.636-1.446-1.033-2.387-1.033c-1.806,0-3.27,1.464-3.27,3.27 c0,0.256,0.029,0.506,0.085,0.745C5.163,5.404,2.753,4.102,1.14,2.124C0.859,2.607,0.698,3.168,0.698,3.767 c0,1.134,0.577,2.135,1.455,2.722C1.616,6.472,1.112,6.325,0.671,6.08c0,0.014,0,0.027,0,0.041c0,1.584,1.127,2.906,2.623,3.206 C3.02,9.402,2.731,9.442,2.433,9.442c-0.211,0-0.416-0.021-0.615-0.059c0.416,1.299,1.624,2.245,3.055,2.271 c-1.119,0.877-2.529,1.4-4.061,1.4c-0.264,0-0.524-0.015-0.78-0.046c1.447,0.928,3.166,1.469,5.013,1.469 c6.015,0,9.304-4.983,9.304-9.304c0-0.142-0.003-0.283-0.009-0.423C14.976,4.29,15.531,3.714,15.969,3.058z"/>
                </svg>
              </span>

              <span class="username">jandot</span>
            </a>
          </li>
          
        </ul>
      </div>

      <div class="footer-col  footer-col-3">
        <p class="text">From data to insight – Putting the human back in the loop of data analysis
</p>
      </div>
    </div>

  </div>

</footer>


  </body>

</html>
