<h1 id="rdbms-and-sql">RDBMS and SQL</h1>

<p>In the following exercises, we will be using <code>sqlite3</code>. It is a lightweight system for dealing with RDBMS databases that can be stored in normal files on a file system. This is in contrast to systems like MySQL which are more widely used, but require to run a dedicated database server. Consider <code>sqlite</code> is like Microsoft Access, where the whole database itself is nothing more than a file in your directory.</p>

<p>A tutorial on <code>sqlite</code> can be found <a href="http://zetcode.com/db/sqlite/">here</a> and instruction videos can be found on Youtube.</p>

<p>We refer to the <a href="datasets.html">description of the datasets</a> for more information about what each of them contains.</p>

<p>For this exercise session, we will consider</p>

<ul>
  <li>data modeling</li>
  <li>data loading</li>
  <li>data querying</li>
</ul>

<p><strong>CAUTION</strong>: When using sqlite3, make sure that you know whether you are on the <strong>unix command line</strong> (the place you already know and love: for running commands such as <code>ls</code>, <code>cd some_directory</code>, <code>less some_file.txt</code>, etc), and the <strong>sqlite3 command line</strong> (where you can issue SQL commands such as <code>SELECT column1, column2 FROM some_table;</code>). The <strong>unix command line</strong> is similar to this:</p>

<pre><code>jaerts@ip-172-31-4-160:~$
</code></pre>

<p>The <strong>sqlite3 command line</strong> looks like this:</p>

<pre><code>sqlite&gt; 
</code></pre>

<p>To go from the unix command line to the sqlite3 command line, type:</p>

<pre><code>sqlite3 name_of_my_database.db
</code></pre>

<p>To exit the sqlite3 command line and return to the unix command line, type:</p>

<pre><code>.quit
</code></pre>

<p>Do <em>not</em> try to run SQL commands on the unix command line; do <em>not</em> try to run unix commands on the sqlite3 command line…</p>

<p>Example use for sqlite3 if we want to create a new database called <code>students</code>. Here’s a sample session (obviously do <strong>not</strong> type in the text of the prompt itself…):</p>

<pre><code>jaerts@ip-172-31-4-160:~$ sqlite3 students.db
sqlite&gt; CREATE TABLE individuals (id INTEGER AUTO_INCREMENT, s_number STRING, first_name STRING, last_name STRING);
sqlite&gt; CREATE TABLE courses (id INTEGER AUTO_INCREMENT, course_number STRING, name STRING);
sqlite&gt; INSERT INTO individuals (s_number, first_name, last_name) VALUES ('s123456','Tom','Thompson');
sqlite&gt; INSERT INTO individuals (s_number, first_name, last_name) VALUES ('s987654','John','Jones');
sqlite&gt; SELECT * FROM individuals;
sqlite&gt; .quit
jaerts@ip-172-31-4-160:~$
</code></pre>

<p>When on the <code>sqlite3</code> prompt, type <code>.help</code> for a list of <code>sqlite3</code>-specific commands (such as <code>.quit</code>).</p>

<h2 id="what-and-how-to-hand-in">What and how to hand in?</h2>

<p>The three datasets each have a set of questions related to it. The questions to answer are put in <strong>bold</strong>. We ask you to prepare a <strong>PDF</strong> document with the answers to the questions and upload it to the correct location on Toledo.</p>

<p>There are two types of questions:</p>

<ol>
  <li>Questions that require you to write a respone in text. For instance: <em>Is this dataset normalized?</em> For this kind of datasets you are asked to answer not only <em>yes</em> or <em>no</em>, but rather <em>This dataset is/is not normalized</em>. </li>
  <li>Questions that require you to write a SQL statement (code). In this case, <strong>copy the code as well as the output in your assignment report</strong>.</li>
</ol>

<p>The report should be in PDF, but you are free to choose any intermediate word editor (Word, Pages, …) as long as you convert to PDF in order to hand it in. <strong>Your report can at most have 2 pages!</strong></p>

<h2 id="beers">Beers</h2>

<h3 id="sql">SQL</h3>

<p>Creating the table to hold the data.</p>

<pre><code>CREATE TABLE beers(id INTEGER, beer TEXT, type TEXT, alc FLOAT, brewery TEXT);
</code></pre>

<p>Select the appropriate field delimiter.</p>

<pre><code>.separator ","
</code></pre>

<p>Import the data. This is easiest from the CSV data.</p>

<pre><code>.import /mnt/bioinformatics_leuven/i0u19a/data/beer/beers.csv beers
</code></pre>

<p>Review whether this was successful and whether the result makes sense. Don’t forget the <code>;</code> symbol at the end of the line! The first entry in the data is the header, which should be removed.</p>

<p><strong>Get the top-10 of breweries based on the number of beers they produce.</strong></p>

<p>Why is AB Inbev not in the top 10? List all beers that are brewed by a brewery that contains the word ‘Inbev’. </p>

<p><strong>Correct that in the database, giving all entries related to AB Inbev the same name.</strong></p>

<p>Suddenly, probably as expected, AB Inbev ranks highest.</p>

<p><strong>How many times does it occur? What is the top-10 now?</strong></p>

<h3 id="data-model">Data Model</h3>

<p><strong>Is this dataset normalized?</strong></p>

<p><strong>Is this the proper way to deal with the data, given the questions above?</strong></p>

<p><strong>How would you do it differently?</strong></p>

<h2 id="drugdb">Drugdb</h2>

<h3 id="sql-1">SQL</h3>

<p>We mentioned that one of the two datafiles contains a subset of the parameters of the large one. Check this explicitely. Make sure you save the routine for importing the data into <code>sqlite</code>.</p>

<p>Remember that there are two source files:</p>

<ul>
  <li><code>AMM_det_H</code> contains the active substances in the doses they have been granted permission to use in drug compounds.</li>
  <li><code>AMM_H</code> contains the drug compounds that can be sold on the market.</li>
</ul>

<p>Is this data normalized? What is the key that joins both datasets together? Does it make sense to organize the data in this way?</p>

<p>First import the full database.</p>

<p><code>
CREATE TABLE drugs1 (
    nr INTEGER,
    cti INTEGER,
    mpname TEXT,
    mah TEST,
    Registratienummer TEXT,
    registdate TEXT,
    generic INTEGER,
    packsize TEXT,
    supplyproblem TEXT,
    PharmFormNl TEXT,
    PharmFormFr TEXT,
    PharmFormDe TEXT,
    PharmFormEn TEXT,
    PackNl TEXT,
    PackFr TEXT,
    PackDe TEXT,
    PackEn TEXT,
    CommNl TEXT,
    CommFr TEXT,
    CommDe TEXT,
    CommEn TEXT,
    DelivNl TEXT,
    DelivFr TEXT,
    DelivDe TEXT,
    DelivEn TEXT,
    ActSubsts TEXT,
    GenNl TEXT,
    GenFr TEXT,
    GenDe TEXT,
    GenEn TEXT,
    GenPK TEXT
    );
.separator ","
.import /mnt/bioinformatics_leuven/i0u19a/data/drugdb/AMM_H.csv drugs1
</code></p>

<p>Do the same for the subset database.</p>

<p><code>
CREATE TABLE drugs2 (
    nr INTEGER,
    cti INTEGER,
    ActSubstName TEXT,
    unit TEXT,
    dosis TEXT,
    DateNew DATE
);
.separator ","
.import /mnt/bioinformatics_leuven/i0u19a/data/drugdb/AMM_det_H.csv drugs2
</code></p>

<p><strong>What are the dimensions of both tables?</strong></p>

<p><strong>Which one is bigger? Join the two data tables and look at some entries.</strong></p>

<p>Note: to make life easier, you can change the <em>output mode</em> of sqlite: <code>.mode column</code> for instance.</p>

<p><strong>What is the top-10 of compounds with the most active substances?</strong></p>

<p>What type of compounds/products are in the top-10? Is this normal?</p>

<p><strong>Which companies have compounds on the market with more than 10 active substances? Put this information in a table.</strong></p>

<p><strong>From this data, create a table that shows for every of the companies in the table, how many complex compounds they have on the market.</strong></p>

<h3 id="data-model-1">Data Model</h3>

<p><strong>Is this dataset normalized?</strong></p>

<p><strong>Is this the proper way to deal with the data, given the questions above?</strong></p>

<p><strong>How would you do it differently?</strong></p>

<h2 id="genotypes">Genotypes</h2>

<h3 id="sql-2">SQL</h3>

<p>Look at the data. Is this it normalized? Is this a handy form for querying the data?</p>

<p>We will focus first on the limited dataset for the first sample.</p>

<p>Create the table.</p>

<p>```
CREATE TABLE geno(
   chrom CHARACTER,
   pos INTEGER,
   id TEXT,
   ref CHARACTER,
   alt CHARACTER,
   qual TEXT,
   filter TEXT,
   info TEXT,
   format TEXT,
   HG00096 TEXT);</p>

<p>.mode tabs
.separator “\t”
.import /mnt/bioinformatics_leuven/i0u19a/data/genotypes/chr1-0-100000_HG000096.tsv geno
```</p>

<p><strong>How many mutations are known for this genomic region on chromosome 1?</strong></p>

<p><strong>How many of these are there in this region for sample HG00096?</strong></p>

<p>If you don’t know the way this information is encoded in the vcf file, refer to the web for more information. What is a good way to get all the mutations for this sample? Think about it.</p>

<h3 id="data-model-2">Data Model</h3>

<p><strong>Is this dataset normalized?</strong></p>

<p><strong>Is this the proper way to deal with the data, given the questions above?</strong></p>

<p><strong>How would you do it differently?</strong></p>
